modules/vaos/app/services/vaos/v2/appointments_service.rb

```diff
@@ -199,34 +199,26 @@ def fetch_clinic_appointments(start_time, end_time, statuses)
         get_appointments(start_time, end_time, statuses)[:data].select { |appt| appt.kind == 'clinic' }
       end
 
-      def prepare_appointment(appointment)
+      def prepare_appointment(appt)
         # for CnP, covid, CC and telehealth appointments set cancellable to false per GH#57824, GH#58690, ZH#326
-        if cnp?(appointment) || covid?(appointment) || cc?(appointment) || telehealth?(appointment)
-          set_cancellable_false(appointment)
-        end
+        set_cancellable_false(appt) if cnp?(appt) || covid?(appt) || telehealth?(appt) || (cc?(appt) && booked?(appt))
```

Comment:
It may be more readable if you put all the || statements after the if conditional into their own method

Reply:
Thank you for the tip, I need to rebuild this work into a new PR so I will carry over this feedback to the new one